### PR TITLE
Enable feature toggle status App and Infrastructure

### PR DESCRIPTION
**Pull request by opan**

### CODE REVIEW

File: app/controllers/api/apps_controller.rb [hunk 1]
```diff
@@ -3,6 +3,15 @@ class Api::AppsController < Api::BaseController
```
Comment (giosakti) [inline app/controllers/api/apps_controller.rb]: just return 404, and you also have to check if app group is active here
I think it's better if you have another method in BaritoApp model called `#available?` it checks for app.active? and infra active?

File: app/controllers/api/infrastructures_controller.rb [hunk 2]
```diff
@@ -1,9 +1,18 @@
 class Api::InfrastructuresController < Api::BaseController
-  skip_before_action :authenticate_token, only: [:profile_by_cluster_name, :authorize_by_username]
+  skip_before_action :authenticate_token
 
   def profile_by_cluster_name
     @infrastructure = Infrastructure.find_by(
       cluster_name: params[:cluster_name])
+
```
Comment (giosakti) [inline app/controllers/api/infrastructures_controller.rb]: please return 404 and only say `Infrastructure not found`

File: app/controllers/api/infrastructures_controller.rb [hunk 3]
```diff
@@ -32,10 +41,23 @@ def authorize_by_username
     @infrastructure = Infrastructure.
       find_by_cluster_name(params[:cluster_name])
 
-    if @infrastructure.blank? 
-      render json: { success: false, errors: ["Unauthorized: infrastructure #{params[:cluster_name]} is not exists"], code: 401 }, status: :unauthorized and return 
+    if @infrastructure.blank?
```
Comment (giosakti) [inline app/controllers/api/infrastructures_controller.rb]: why don't we combine these 2 conditionals like the above

File: app/controllers/api/infrastructures_controller.rb [hunk 4]
```diff
@@ -32,10 +41,23 @@ def authorize_by_username
...
+      render json: {
+        success: false, 
+        errors: ["Unauthorized: infrastructure #{params[:cluster_name]} is not exists"],
+        code: 401
+      }, status: :unauthorized and return 
     end
 
+    if @infrastructure.status == Infrastructure.statuses[:inactive]
```
Comment (giosakti) [inline app/controllers/api/infrastructures_controller.rb]: why don't use `infrastructure.active?`
also please only return 404

File: app/controllers/api/apps_controller.rb (hunk 1 above)
Comment (opan) [reply app/controllers/api/apps_controller.rb]: Nice suggestion. Will do it

File: app/controllers/api/infrastructures_controller.rb (hunk 4 above)
Comment (opan) [reply app/controllers/api/infrastructures_controller.rb]: Forgot to replace this with that method. This is written before `infrastructure.active?` method is created

**Pull request merged**